src/chapters/setup-intro-data-types/more-data-types.rst

```diff
@@ -2,258 +2,321 @@ More Data Types
 ================
 
 Class Types
------------
+------------
 
-In addition to the types introduced so far - primitives and their
-objectified counterparts - any class in Java defines a type. Classes and
-objects are conceptually the same as in Python: A class is a template
-for creating objects. We’ll have much more to say about classes and
-objects, but for now you need to be comfortable seeing the basic syntax
-of class types and class creation.
+.. index:: ! class
 
-If I have a class ``Cat`` with a constructor that takes no arguments, I
-can declare and create a new instance of ``Cat`` using its constructor.
-In Python, we did this as follows:
+In Java, a **class** is a template for creating objects. In addition to the
+data types introduced so far, any class in Java also defines a type. We’ll have
+much more to say about classes and objects, but for now you need to recognize
+the basic syntax of class types and class creation.
 
-.. code:: python
+If we have a class ``Cat``, we can declare and create an instance of ``Cat``
+using the ``new`` keyword:
 
-   # Python
-   my_cat = Cat()
-
-And the Java version is:
-
-.. code:: java
+.. sourcecode:: java
 
    Cat myCat = new Cat();
 
-Each of these statements creates a new variable that is initialized to
-hold a new ``Cat`` object. Note that in Java, we must declare the
-variable’s type. Also note that we precede the constructor with the
-``new`` keyword. And, of course, the Java example ends with a
-semi-colon.
+#. ``Cat myCat`` declares the variable ``myCat`` and sets it to be of type
+   ``Cat``.
+#. ``= new Cat()`` initializes the variable with a new ``Cat`` object.
+#. Any arguments that are required to build the new ``Cat`` object must be
+   included within the parentheses.
+
+.. index:: ! reference type
 
-Variables and parameters that are of the type of a class are said to be
-of **reference type** (in contrast to **primitive type**). In plain
-English, we would say of the Java example: “``myCat`` is a reference
-variable of type ``Cat``.”
+Just like variables or parameters can be declared as an ``int`` or ``String``
```

Comment:
i would leave out the 'or parameters' part of this sentence

Reply:
Fixed.